### PR TITLE
refactor: simplify logic in get-challenges

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- refactor: validate earlier in build pipeline
- refactor: simplify certification creation
- refactor: find if superblock is audited once

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: curriculum/get-challenges.js [hunk 1]
```diff
@@ -217,48 +221,24 @@ async function buildChallenges({ path: filePath }, curriculum, lang) {
   }
   const { meta } = challengeBlock;
   const isCert = path.extname(filePath) === '.yml';
+  const englishPath = path.resolve(__dirname, CHALLENGES_DIR, lang, filePath);
```
Comment (naomi-lgbt) [inline curriculum/get-challenges.js:224]: Is this an accurate name? Is `lang` always going to be `english` here?

Comment (ojeytonwilliams) [reply curriculum/get-challenges.js:224]: :man_facepalming:

Comment (naomi-lgbt) [reply curriculum/get-challenges.js:224]: Haha it's okay, I speak "facepalm" - goodness knows I have enough of those moments myself

**Review by naomi-lgbt — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Sorry, @naomi-lgbt, I was struggling with this one and thought I would stop with what I had, but didn't check carefully enough.

What I'm trying to do is untangle the logic a bit. For instance, we're doing things (like calculating if a block is audited) at the challenge level, rather than in the block. It doesn't make sense to me. I think that can wait until round 3 :face_exhaling: 

Anyways, thanks for the review!

**Review by raisedadead — APPROVED** [review]

**Pull request merged**